File: 1544-make-the-string-great/1544-make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []

        for char in s:
            if stack and stack[-1].isupper() and stack[-1].lower() == char:
                stack.pop()
            elif stack and stack[-1].islower() and stack[-1].upper() == char:
                stack.pop()
            else:
                stack.append(char)

        return "".join(stack)

File: 1544-make-the-string-great/1544-make-the-string-great.rs
impl Solution {
    pub fn make_good(s: String) -> String {
        let mut stack: Vec<char> = Vec::new();
        
        for c in s.chars() {
            if let Some(top) = stack.last() {
                if top.is_uppercase() && top.to_ascii_lowercase() == c {
                    stack.pop();
                } else if top.is_lowercase() && top.to_ascii_uppercase() == c {
                    stack.pop();
                } else {
                    stack.push(c);
                }
            } else {
                stack.push(c);
            }
        }
        
        stack.iter().collect()
    }
}
